### PR TITLE
Use Guice 4.0 like Jenkins core

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Fixes a compatibility issue while running tests with the DockerRule

@reviewbybees @jglick

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -72,7 +72,7 @@
         <dependency>
             <groupId>com.google.inject</groupId>
             <artifactId>guice</artifactId>
-            <version>3.0</version>
+            <version>4.0-beta</version>
```
Comment (jglick) [inline pom.xml]: Jenkins core currently uses `4.0` (not beta).

Comment (oleg-nenashev) [inline pom.xml]: Yep, beta came from the parent PoM

**Comment by oleg-nenashev** [comment]
> Did you check the impact on acceptance-test-harness, if it were to pick up a new version of this library?

I didn't. Probably it requires a downstream PR in order to become a subject for consideration

File: pom.xml (hunk 1 above)
Comment (jglick) [inline pom.xml]: Since 4.0 is released, I see no reason to be using a beta build now.

**Comment by olivergondza** [comment]
The idea looks good to me.

**Pull request merged**